packages/pwa/app/utils/url.js

```diff
@@ -185,3 +211,36 @@ export const removeQueryParamsFromPath = (path, keys) => {
 
     return `${pathname}${paramStr && '?'}${paramStr}`
 }
+
+/**
+ * Rebuild the path by adding/removing locale/site query params based on url config
+ * @param {string} url - based url of the output url
+ * @param {object} options - object that contains values of url config key
+ * @return {string} - an output url
+ *
+ * @example
+ * //pwa-kit config
+ * url {
+ *    locale: "query_param"
+ * }
+ * buildPathWithLocaleAndSiteParams('/women/dresses/', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /women/dresses/?locale=en-GB
+ *
+ */
+export const buildPathWithLocaleAndSiteParams = (url, options = {}) => {
```

Comment:
Using `basename` on SDK Routers saves us the logic of modify the routes, however, we still have a logic to handle the case when the locale/siteId is configured as a `query_param`. For example, we want all routes to have be something like this `/women/dresses/?locale=en-GB` across the app. This means we need to use this util on some url builders utils like `productUrlBuilder`, `searchUrlBuilder`, `categoryUrlBuilder`, `Link` component, `useNavigation` hook, `history.push` from `_app` component (bc we can't use `useNavigation` here). 

One idea I had is to create a hook to modify the history.location.search object to preserve some fixed params (eg. locale, siteId) so we don't have to reuse the util function inside react components but it could create more complications to the pwa. 🤔 

I am open to any idea that can simplify the logic here. Thoughts?

Reply:
Will this help to preserve the query parameters? https://stackoverflow.com/questions/43613140/how-to-preserve-query-parameters-in-react-router-v4

Reply:
Similar to my other comment, the name for this function will probably make it hard to expand on. Example, me might want to have configurability on the currency.

Reply:
Thanks for the comments folks, I renamed it to be `buildPathWithUrlConfigParams`  if that makes more sense

Reply:
@vmarta, I've added a hook try to preserve the query param [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/183/files#diff-975bc7b1a69af6abc2963185de1f75bf1e79968b5817d91db0aeee1f328f4e36R1). This is likely to be a temporary hook as we dont have access `intl` in `App` component. When we finish our work on moving `intl` to `AppConfig`, we can safely use `useNavigation` hook, and it will preserve the query param. 

What are your thoughts on this? 🤔